frame/ethereum/src/lib.rs

```diff
@@ -632,16 +637,42 @@ impl<T: Config> Pallet<T> {
 			),
 		};
 
-		let receipt = ethereum::Receipt {
-			state_root: match reason {
-				ExitReason::Succeed(_) => H256::from_low_u64_be(1),
-				ExitReason::Error(_) => H256::from_low_u64_le(0),
-				ExitReason::Revert(_) => H256::from_low_u64_le(0),
-				ExitReason::Fatal(_) => H256::from_low_u64_le(0),
-			},
-			used_gas,
-			logs_bloom: status.clone().logs_bloom,
-			logs: status.clone().logs,
+		let receipt = {
+			let status_code: u8 = match reason {
+				ExitReason::Succeed(_) => 1,
+				_ => 0,
+			};
+			let logs_bloom = status.clone().logs_bloom;
+			let logs = status.clone().logs;
+			let cumulative_gas_used = if let Some((_, _, receipt)) = pending.last() {
+				match receipt {
+					Receipt::Legacy(d) => d.used_gas.saturating_add(used_gas),
```

Comment:
```suggestion
					Receipt::Legacy(d) | Receipt::EIP2930(d) | Receipt::EIP1559(d) => d.used_gas.saturating_add(used_gas),
```

This works because the inner `d` is of the same type.